packages/server/src/app.ts

```diff
@@ -205,3 +206,12 @@ export async function shutdownApp(): Promise<void> {
     rmSync(binaryStorage.replace('file:', ''), { recursive: true, force: true });
   }
 }
+
+const loggingMiddleware = (req: Request, res: Response, next: NextFunction):void =>{
+  const start = new Date();
+  next();
+  const afterNext = new Date().getTime(); // Record the time after next() completes
+  const totalTime = afterNext - start.getTime(); // Calculate the time taken including the time spent in next()  
+  globalLogger.log('INFO', 'logging data', {requestReceivedTimestamp:start,requestMethodAndPath: `${req.method} ${req.path}`,timeTaken: `${totalTime} ms`, IP_address: req.ip, status: res.statusCode, userAgentString: req.get('User-Agent')});
```

Comment:
Instead of using the global logger here, it would be better to use the request-specific one in order to get the request ID and trace ID attached:
```ts
const ctx = getRequestContext();
ctx.logger.info(/* ... */);
```

Additionally, you can access the current user's profile information through the request context:
```ts
let userProfile: string | undefined;
if (ctx instanceof AuthenticatedRequestContext) {
  userProfile = ctx.profile.reference;
}
```

A few other minor notes:
- `INFO` is the correct level for this log, but the log message should be more meaningful: consider changing this from "logging data" to something like "Request served" to describe what the log entry is recording
- It's a bit hard to scan this long line with all the data properties, could you put each property on its own line?
- Since this log is going to be written many times and will likely make up a significant portion of overall log volume, I would like to try making the property names shorter to save space.  I know this might seem nitpicky, but tens of bytes can add up fast at scale!  I would suggest the following property names: `receivedAt`, `method`, `path`, `duration`, `ip`, `status`, and `ua` for brevity

Reply:
Hi @mattwiller, Thank for your feedback. I have changed the code.